OpenLaTeXStudio/EditorModule/src/latexstudio/editor/settings/ApplicationSettings.java

```diff
@@ -107,12 +117,76 @@ public boolean getLineWrapStatus(){
         return Boolean.parseBoolean(getProperty(LINEWRAP_STATUS));
     }
     
-    public void setAutoCompleteStatus(boolean autoCompleteStatus){
-        pcs.firePropertyChange(AUTOCOMPLETE_STATUS, getAutoCompleteStatus(), autoCompleteStatus);
-        setProperty(AUTOCOMPLETE_STATUS, String.valueOf(autoCompleteStatus));
+    public void registerSettingListeners(Object o) {
+        for(Method m : o.getClass().getMethods()) {
+            for(Annotation a : m.getAnnotations()) {
+                if( a instanceof SettingListener ) {
+                    Setting s = ((SettingListener) a).setting();
+                    //TODO 20160323 Type validation
+                    settingListeners.get(s).add(Pair.of(o,m));
+                    try {
+                        m.invoke(o, getSettingValue(s));

```

Comment:
This line provides convenient enhancement of the registration with a first "update" of the setting with the current value.
Also, although I Javadoc'ed the annotation now, methods not fulfilling the contract should be ignored here.